src/java.base/share/classes/java/lang/invoke/MethodType.java

```diff
@@ -415,7 +409,6 @@ public static MethodType genericMethodType(int objectArgCount) {
      */
     public MethodType changeParameterType(int num, Class<?> nptype) {
         if (parameterType(num) == nptype)  return this;
-        checkPtype(nptype);
```

Comment:
`nptype` is never void but  what  about the check if `nptype` is not null?

Reply:
Other methods that delegate to `makeImpl` aren't doing up-front validation, so this change was made to get things more in line. It might be good to spell out that `makeImpl` does these checks for all its callers, though. (The `makeImpl` fast-path that execute before the validation can never return an invalid MethodType)

Reply:
Generally we have a public API implementation to check the arguments upfront for readability.  In particular for this case, the validation cost is negligible and removing the validation makes the code unclear where the validation is done.  I prefer to keep the validation there.   It should check that `nptype` is non-null and not `void.class`.

Reply:
Ah, I mis-read your comment.   I now see that `makeImpl` does the argument validation.  Adding the comment would be helpful.